dll/cpl/usrmgr/usrmgr.c

```diff
@@ -15,6 +16,22 @@ static LONG APIENTRY UsrmgrApplet(HWND hwnd, UINT uMsg, LPARAM wParam, LPARAM lP
 
 HINSTANCE hApplet = 0;
 
+VOID GetDlgItemTextAlloc(HWND hwndDlg, INT nDlgItem, LPTSTR *ppsz)
+{
+    INT nLength = GetWindowTextLength(GetDlgItem(hwndDlg, nDlgItem));
+    *ppsz = HeapAlloc(GetProcessHeap(), 0, (nLength + 1) * sizeof(TCHAR));
+    if (*ppsz)
+        GetDlgItemText(hwndDlg, nDlgItem, *ppsz, nLength + 1);
+}
```

Comment:
For example:
```suggestion
LPTSTR GetDlgItemTextAlloc(HWND hwndDlg, INT nDlgItem)
{
    INT nLength = GetWindowTextLength(GetDlgItem(hwndDlg, nDlgItem));
    LPTSTR psz = HeapAlloc(GetProcessHeap(), 0, (nLength + 1) * sizeof(TCHAR));
    if (psz)
        GetDlgItemText(hwndDlg, nDlgItem, psz, nLength + 1);
    return psz;
}
```

Reply:
Also, what to do if the window text length is zero? Should we always allocate a "one-character"-sized string just to store the NULL char? Or should we return NULL?

Reply:
Fixed.

> Also, what to do if the window text length is zero?

This data is used in `dll/win32/netapi32/user.c`. `NULL` data is invalid. The empty string is valid data.